pkg/porter/credentials.go

```diff
@@ -106,22 +107,91 @@ func (p *Porter) GenerateCredentials(opts CredentialOptions) error {
 	if err != nil {
 		return err
 	}
-	bundle, err := p.CNAB.LoadBundle(opts.CNABFile)
 
+	_, err = p.generateNewCredentialSet(opts.CNABFile, opts.Name, opts.Silent, opts.DryRun)
+	return err
+}
+
+func (p *Porter) chooseCredentialSet(CNABFile string) (string, error) {
+	var credSetNames []string
+	credSets, err := p.Credentials.ReadAll()
 	if err != nil {
-		return err
+		return "", errors.Wrap(err, "failed to read exisiting credential sets")
+	}
+
+	shouldGenerateCred := false
+	var selectedOption string
+	if len(credSets) > 0 {
+		for _, credSet := range credSets {
+			credSetNames = append(credSetNames, credSet.Name)
+		}
+
+		selectCredPrompt := &survey.Select{
+			Message: "Choose a set of credentials to use while installing this bundle",
+			Options: append(credSetNames, generateCredCode, quitGenerateCode),
+			Default: credSetNames[0],
+		}
+		survey.AskOne(selectCredPrompt, &selectedOption, nil)
+
+		switch selectedOption {
+		case generateCredCode:
+			shouldGenerateCred = true
+		case quitGenerateCode:
+			return "", errors.New("Credentials are mandatory to install this bundle")
+		default:
+			return selectedOption, nil
+		}
+	} else {
+		shouldGenerateCredPrompt := &survey.Confirm{
+			Message: "No credential identifier given. Generate one ?",
+		}
+		survey.AskOne(shouldGenerateCredPrompt, &shouldGenerateCred, nil)
+	}
+
+	if !shouldGenerateCred {
+		return "", errors.New("Credentials are mandatory to install this bundle")
 	}
-	name := opts.Name
-	if name == "" {
-		name = bundle.Name
+
+	return p.generateNewCredentialSet(CNABFile, "", false, false)
+}
+
+func (p *Porter) generateNewCredentialSet(CNABFile string, credIdentifierName string, silent bool, dryRun bool) (string, error) {
+	bundle, err := p.CNAB.LoadBundle(CNABFile)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to load bundle while generating credentials")
 	}
+
+	if credIdentifierName == "" {
+		if !silent {
```

Comment:
Just a suggestion but it's usually easier to read the positive case `if silent` than the inverse, especially when there is an else clause. Since you do have an action for when silent is true, it would be easier to follow if this was `if silent {  // logic for silent case } else { // logic for not silent }`.

Up to you if you want to switch it up.

Reply:
couldn't agree more